decidim-proposals/spec/system/proposals_spec.rb

```diff
@@ -643,4 +643,57 @@
       it_behaves_like "an uncommentable component"
     end
   end
+
+  describe "viewing mode for proposals" do
+    let!(:proposal) { create(:proposal, :evaluating, component:) }
+
+    context "when participants interact with the proposal view" do
+      it "provides an option for toggling between list and grid views" do
+        visit_component
+        expect(page).to have_css("use[href*='layout-grid-fill']")
+        expect(page).to have_css("use[href*='list-check']")
+      end
+    end
+
+    context "when participants are viewing a grid of proposals" do
+      it "shows a grid of proposals with images" do
+        visit_component
+        find("a[href*='view_mode=grid']").click
+        expect(page).to have_css(".card__grid-grid")
+        expect(page).to have_css(".card__grid-img img, .card__grid-img svg")
```

Comment:
I would take this test and check that there's no grid before clicking the button (checking default mode), this do this check, and then revisit the component and check that the grid is still there (so we check the session storage).

Reply:
Thank you for the feedback! I have updated the tests as per your suggestion